baker/algolia/utils/pages.ts

```diff
@@ -105,15 +106,20 @@ async function generateWordpressRecords(
     const getPostTypeAndImportance = (
```

Comment:
I wanted to rewrite the `PageType` type to use the `OwidGdocType` enum so that it would be compatible with `getCanonicalUrl` which we now need to be able to link to data insights correctly on the search page, so I had to refactor these 3 functions that each had a pattern match statement.